app/models/user/activity.rb

```diff
@@ -0,0 +1,59 @@
+class User::Activity < ApplicationRecord
+  belongs_to :user
+  belongs_to :track, optional: true
+
+  before_create do
+    self.version = latest_i18n_version
+    self.uniqueness_key = "#{user_id}|#{type_key}|#{guard_params}"
+    self.grouping_key = "#{user_id}|#{grouping_params}"
+    self.occurred_at = Time.current unless self.occurred_at
+  end
+
+  def text
+    I18n.t("user_activities.#{i18n_key}.#{version}", i18n_params).strip
+  end
+
+  def widget
+    nil
+  end
+
+  def type_key
+    type.split("::").last.underscore.split("_activity").first.to_sym
+  end
+
+  # This maps
+  # {discussion: Solution::MentorDiscussion.find(186)}
+  # to
+  # {discussion: "gid://exercism/Solution::MentorDiscussion/186"}
+  #
+  # Any non-object params are left as the were passed in.
+  def params=(hash)
+    self[:params] = hash.transform_values do |v|
+      v.respond_to?(:to_global_id) ? v.to_global_id.to_s : v
+    end
+  end
+
+  # This reverses params= to explode back out
+  # {discussion: "gid://exercism/Solution::MentorDiscussion/186"}
```

Comment:
What is the use of `gid://`, as in: where it that used?

Reply:
https://github.com/rails/globalid#global-id---reference-models-by-uri

It's used by Rails to serlalized and reload objects based on their ids.